### DISCUSSION TITLE
How to use view as parameter ?

### DISCUSSION THREAD

**Discussion by Louis-PAGNIER** [skiptools/skip#136] [Q&A] [answered]
In the project that I port with Skip, I often use the `@ViewBuilder` syntax in struct.
However, when I run it with Skip, I have the following errors:
![image](https://github.com/skiptools/skip/assets/55755544/57356a8c-ca49-4de3-b769-7bb72b0a4db4)

Is there an alternative way to write it for Skip ?

**Comment by marcprux** [comment, marked as answer]
We don't support using `@ViewBuilder` as a property of views, but you can use a closure returning a View instead. E.g.:

```swift
struct RedBorderView<V: View> : View {
    let content: () -> V

    var body: some View {
        content()
            .border(Color.red, width: 5.0)
    }
}
```

And then call:

```swift
RedBorderView {
    Text("Hello Red!")
}
```

Hopefully that is sufficient for your needs. If not, please let us know.

**Reply by Louis-PAGNIER** [reply]
Thank you, it's perfect.

**Reply by aabewhite** [reply]
This should now be fixed in skip 0.8.50, and you can use @ViewBuilder without turning your var into a closure. Use File->Packages->Update to Latest to get the latest transpiler with the fix

**Reply by Louis-PAGNIER** [reply]
Thank you!
